extensions/ql-vscode/src/remote-queries/variant-analysis-content-provider.ts

```diff
@@ -0,0 +1,26 @@
+import { TextDocumentContentProvider, Uri } from 'vscode';
+import { URLSearchParams } from 'url';
+import { showAndLogWarningMessage } from '../helpers';
+import { SHOW_QUERY_TEXT_MSG } from '../query-history';
+import { VariantAnalysisManager } from './variant-analysis-manager';
+
+export const createVariantAnalysisContentProvider = (variantAnalysisManager: VariantAnalysisManager): TextDocumentContentProvider => ({
+  async provideTextDocumentContent(uri: Uri): Promise<string | undefined> {
+    const params = new URLSearchParams(uri.query);
+
+    const variantAnalysisIdString = params.get('variantAnalysisId');
+    if (!variantAnalysisIdString) {
+      void showAndLogWarningMessage('No variant analysis ID provided');
```

Comment:
Maybe make these error messages slightly more verbose 🤷 
Otherwise I'm not sure it'll be clear what they mean when they pop up.
```suggestion
      void showAndLogWarningMessage('Unable to show query text. No variant analysis ID provided.');
```